status/status_test.go

```diff
@@ -419,17 +418,24 @@ func (s) TestStatus_ErrorDetails_Fail(t *testing.T) {
 		},
 	}
 	for _, tc := range tests {
-		got := tc.s.Details()
-		if !cmp.Equal(got, tc.i, cmp.Comparer(proto.Equal), cmp.Comparer(equalError)) {
-			t.Errorf("(%v).Details() = %+v, want %+v", str(tc.s), got, tc.i)
+		details := tc.s.Details()
```

Comment:
You should first ensure `len(details) == len(tc.want)` here or it could panic.

Reply:
that's right! thanks for catching that